test/e2e/specs/editor/various/preview.spec.js

```diff
@@ -0,0 +1,362 @@
+		await editorPage.keyboard.press( 'Tab' );
+
+		// Save the post as a draft.
+		await editorPage.click( 'role=button[name="Save draft"i]' );
+		await editorPage.waitForSelector(
+			'role=button[name="Dismiss this notice"] >> text=Draft saved'
+		);
+
+		// Open the preview page.
+		const previewPage = await editor.openPreviewPage( editorPage );
+		await previewPage.waitForSelector( 'role=heading[level=1]' );
+
+		// Title in preview should match input.
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		expect( await previewTitle.textContent() ).toEqual( 'aaaaa' );
+
+		// Return to editor.
+		await editorPage.bringToFront();
+
+		// Append bbbbb to the title, and tab away from the title so blur event is triggered.
+		await editorPage.focus( 'role=textbox[name="Add title"]' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await editorPage.keyboard.press( 'ArrowRight' );
+		await editorPage.keyboard.type( 'bbbbb' );
+		await editorPage.keyboard.press( 'Tab' );
+
+		// Save draft and open the preview page right after.
+		await editorPage.click( 'role=button[name="Save draft"i]' );
+		await editorPage.waitForSelector(
+			'role=button[name="Dismiss this notice"] >> text=Draft saved'
+		);
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual( 'aaaaabbbbb' );
+
+		await previewPage.close();
+	} );
+
+	// Verify correct preview. See: https://github.com/WordPress/gutenberg/issues/33616
+	test( 'should display the correct preview when switching between published and draft statuses', async ( {
+		editor,
+		page,
+		previewUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Type Lorem in the title field.
+		await editorPage.type( '[aria-label="Add title"]', 'Lorem' );
+
+		// Open the preview page.
+		const previewPage = await editor.openPreviewPage( editorPage );
+		await previewPage.waitForSelector( 'role=heading[level=1]' );
+
+		// Title in preview should match input.
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		expect( await previewTitle.textContent() ).toEqual( 'Lorem' );
+
+		// Return to editor and publish post.
+		await editorPage.bringToFront();
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.locator( 'role=button[name="Close panel"]' ).click();
+
+		// Change the title and preview again.
+		await editorPage.type( '[aria-label="Add title"]', ' Ipsum' );
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual( 'Lorem Ipsum' );
+
+		// Return to editor and switch to Draft.
+		await editorPage.bringToFront();
+		await page.locator( 'role=button[name="Switch to draft"]' ).click();
+
+		// Change the title.
+		await editorPage.type( '[aria-label="Add title"]', ' Draft' );
```

Comment:
I think we can change this too?

```suggestion
		await editorPage.type( 'role=button[name="Add title"i]', ' Draft' );
```

Reply:
So this is interesting. The `role=textbox[name="Add title"]` selector only works when initially adding a title. Once the title is set, it fails to find it. Might it be because the element text content is now different from its aria-label, which is still "Add title"? I did try changing `name` to whatever the title was set to at the time, but that didn't work either (also targeting a text input by its current content feels a bit random 😅 )

Reply:
Oh, I think the name is still `Add title` and the selector should work. The problem is that we're missing the "pressing OK in the dialog" action in Line 186. See my other comment there :)